### PR TITLE
Minor fixes to new block families' docs (merge to newBlockFamilies branch)

### PR DESCRIPTION
**Pull request by anuar2k**
### Contains

Minor fixes to the newBlockFamilies' javadocs as an extra for Block Families Task no. 3.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/world/block/family/FreeFormSupported.java [hunk 1]
```diff
@@ -20,6 +20,10 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * Annotation used to mark if a {@link BlockFamily} supports blocks in freeform shapes.
```
Comment (pollend) [inline engine/src/main/java/org/terasology/world/block/family/FreeFormSupported.java]: Annotation used to mark if a {@link BlockFamily} supports freeform shapes or blocks that have freeform shapes.

File: engine/src/main/java/org/terasology/world/block/family/RegisterBlockFamily.java [hunk 2]
```diff
@@ -21,6 +21,10 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * This annotation is used to mark a {@link BlockFamily} to be registered into the game.
```
Comment (pollend) [inline engine/src/main/java/org/terasology/world/block/family/RegisterBlockFamily.java]: look at RegisterSystem annotation. The behavior is fairly similar. You can provide some examples of the usage.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by pollend** [comment]
refrences: https://github.com/MovingBlocks/Terasology/pull/3149

**Pull request merged**